File: backend/app/schemas/analytics.py
from datetime import date
from typing import List, Dict

from pydantic import BaseModel

class HeatmapData(BaseModel):
    date: date
    count: int

class CategoryStat(BaseModel):
    category: str
    count: int

class PriorityStat(BaseModel):
    priority: str
    count: int

class AnalyticsResponse(BaseModel):
    heatmap: List[HeatmapData]
    by_category: List[CategoryStat]
    by_priority: List[PriorityStat]
    best_time_of_day: str | None = None
    most_frequent_mood: str | None = None

File: frontend/src/types/analytics.ts
export interface HeatmapData {
  date: string
  count: number
}

export interface CategoryStat {
  category: string
  count: number
}

export interface PriorityStat {
  priority: string
  count: number
}

export interface AnalyticsResponse {
  heatmap: HeatmapData[]
  by_category: CategoryStat[]
  by_priority: PriorityStat[]
  best_time_of_day?: string
  most_frequent_mood?: string
}

export interface Badge {
  id: string
  name: string
  description: string
  icon: string
  is_unlocked: boolean
  progress_percentage: number
}
